src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,92 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  addComment: ({
+    postId,
+    name,
+    email,
+    body,
+  }: Omit<Comment, 'id'>) => Promise<void>;
+  postId: number;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  addComment,
+  postId,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [textError, setTextError] = useState(false);
```

Comment:
My assumption it shoud be 2 states instead of 6
```suggestion
const [formFieds, setFormFieds] = useState({
  name: '',
  email: '',
  text: '',
});
const [errorMessage, setErrorMessage] = useState('');
```

Reply:
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/90213755/9398567e-5621-4e11-9234-ec48385923ad)
its a bad idea to create  errorMessage, because we would use it 3 times.

Reply:
and about object, if i change only name, it will rerender all object.  to my mind, its better to use 3 useStates.

Reply:
> and about object, if i change only name, it will rerender all object. to my mind, its better to use 3 useStates.

FYI: You probably meant that we will change the entire object stored in the state, but that's not a problem. we do not render an object. we render the component (jsx that is written below), and re-create every variable (including) functions during the render. Having different states would be useful if you use memorization hooks with different variables in the dependency arrays.